batch/batch/driver/main.py

```diff
@@ -1140,7 +1141,8 @@ async def on_startup(app):
 
     app['task_manager'].ensure_future(periodically_call(60, scheduling_cancelling_bump, app))
 
-    app['task_manager'].ensure_future(periodically_call(15, monitor_system, app))
+    if HAIL_SHOULD_MONITOR_SYSTEM:
+        app['task_manager'].ensure_future(periodically_call(15, monitor_system, app))
```

Comment:
I think there's another way to handle this more broadly without adding complexity to the driver code. In the prometheus config, you can filter and alter labels on metrics that the kubernetes collector adds, like we do [here](https://github.com/hail-is/hail/blob/4834afaf434dfbadc731899b8374937e1dd8ca36/prometheus/prometheus.yaml#L112-L123). Unless there are google quota issues to worry about, I think it would be simpler to always have our services aggregate and display metrics, but just tell prometheus to ignore those whose `__meta_kubernetes_namespace` matches `pr-.*`.